lib/tjbot.js

```diff
@@ -1062,9 +1062,10 @@ TJBot.prototype._normalizeColor = function(color) {
 TJBot.prototype.speak = function(message) {
     this._assertCapability('speak');
 
-    // make sure we're trying to say something
-    if (message == undefined) {
-        throw new Error("TJBot tried to speak an empty message.");
+    // make sure we're trying to say something that we have been trained to understand
+    if (message == undefined || message == '') {
+        message = 'Sorry, I did not understand';
```

Comment:
I'm not sure if this is the right response... when I write `tj.speak("hello")`, it's because I want TJBot to say "hello".  So if I write `tj.speak('')`, he will say "Sorry, I did not understand.". But I'm not sure that's actually what we want.

It might help to understand a little more about the context to determine what the right solution is.  E.g. are you piping the results of `tj.converse()` into `tj.speak()` and finding that `tj.converse()` is returning empty responses?  In that case, maybe the right thing to do is to respond with "Sorry, I did not understand.", but I might argue that behavior should be performed by your recipe, not by the lib.

It's also possible that instead of throwing an error on `tj.speak('')` or `tj.speak(undefined)`, we just fail silently -- don't speak anything at all and just return, without throwing the error.

Curious to hear your thoughts.  :)